### PR TITLE
Feature/js refactoring

### PR DESCRIPTION
**Pull request by achen2401**
address TN-483: extracting inline script where possible
extracting global functions from main.js to utility.js - those probably should be scoped so they don't pollute global name space - will fix when I have time.  
re-factor profile.html template file as it is getting too messy so as to:
create a base template file so other profile templates can inherit from (see profile_base.html)
patient's profile has a template file (i.e. patient_profile.html)
profile template (i.e. profile.html) now contains sections when one is viewing one's own profile
new user profile template file contains sections when one is viewing other user's profile (i.e. user_profile.html)
extracting inline scripts from profile template macros

**Note** this is round 1 of my efforts,  I am not quite done extracting all applicable in-line scripts yet (i.e. consent and org sections and registration invite emails section still contain inline scripts).  Extracting JS from those sections is a bit more complicated.  I will do those when I am back - will make the changes for those in another PR.  I want to get these out so you guys can review.

### CODE REVIEW

File: portal/static/js/utility.js [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+        $(".button--LR").addClass("data-show");
+    };
+
+};
+function getSaveLoaderDiv(parentID, containerID) {
+    var el = $("#" + containerID + "_load");
+    if (el.length === 0) {
+        var c = $("#" + parentID + " #" + containerID);
+        if (c.length > 0) {
+            var snippet = '<div class="load-container">' + '<i id="' + containerID + '_load" class="fa fa-spinner fa-spin load-icon fa-lg save-info" style="margin-left:4px; margin-top:5px" aria-hidden="true"></i><i id="' + containerID + '_success" class="fa fa-check success-icon save-info" style="color: green" aria-hidden="true">Updated</i><i id="' + containerID + '_error" class="fa fa-times error-icon save-info" style="color:red" aria-hidden="true">Unable to Update.System error.</i></div>';
```
Comment (ivan-c) [inline portal/static/js/utility.js]: Please tag for translation (unless this is developer-only)

File: portal/static/js/utility.js [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+    for (var method in methods) {
+        if ( methods.hasOwnProperty(method) && !(method in console) ) { // define undefined methods as best-effort methods
+            console[method] = methods[method];
+        }
+    }
+})();
+
+
+//Promise polyfill - IE doesn't support Promise - so need this
```
Comment (ivan-c) [inline portal/static/js/utility.js]: Can we use CDN-hosted versions of these inline libraries?

File: portal/templates/patient_profile.html [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    &nbsp;{{user.last_name}}
+  {%- elif user.email -%}
+    &nbsp;{{ user.email }}
+  {%- else -%}
+    &nbsp;#{{ user.id}}
+  {%-endif-%}
+{% endblock %}
+{% block profile_content %}
+  <nav id="indexNavBar">
+    <span id="loginAsPatient" data-toggle="modal" data-target="#loginAsModal" class="link"><a data-toggle="tooltip" data-placement="top" title="For 'kiosk' style administration of an assessment">{{ _("Log in as this patient") }}</a></span>
```
Comment (ivan-c) [inline portal/templates/patient_profile.html]: Can you tag the content of the `title` attribute for translation?

File: portal/templates/profile_base.html [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    </div>
+  </div>
+</form>
+{% endblock %}
+{% block footer %}<div class="footer-wrapper right-panel"><div class="flex footer-container"><div class="copyright-container">{%-if providerPerspective== 'true' -%}{{linksHTML(user=current_user)}}{%-else-%}{{linksHTML(user=user)}}{%-endif-%}</div><div class="logo-container">{{logo(True)}}</div></div></div>{% endblock %}
+{% block additional_scripts %}
+<script src="{{ url_for('static', filename='js/procedures.js') }}" async></script>
+{% endblock %}
+{% block document_ready %}
+  $("#mainDiv").addClass("profile");
```
Comment (ivan-c) [inline portal/templates/profile_base.html]: Can this be moved into a separate file (I think it appears in an inline script)?

**Comment by achen2401** [comment]
need to add more changes to this, probably won't be ready for this release.  closing this now.  @ivan-c  Ivan, I will make fixes based on your comment.  You will see them when I re-open this.

**Pull request closed without merging**